src/ast.js

```diff
@@ -572,29 +572,12 @@ function ast_for_with_stmt(c, n0, is_async) {
     body = astForSuite(c, CHILD(n, NCH(n) - 1));
 
     if (is_async) {
-        return new Sk.astnodes.AsyncWith(items, body, LINENO(n0), n0.col_offset);
+        return new Sk.astnodes.AsyncWith(items, body, LINENO(n0), n0.col_offset, n0.end_lineno, n0.end_col_offset);
     } else {
-        return new Sk.astnodes.With(items, body, LINENO(n), n.col_offset);
+        return new Sk.astnodes.With(items, body, LINENO(n), n.col_offset, n.end_lineno, n.end_col_offset);
     }
 }
 
-function astForExecStmt (c, n) {
```

Comment:
Ah sorry for leaving this. We used oldStyleNaming and new_style_naming when moving the ast code so we could see which functions we had done. 😊

Reply:
@meredydd what do you think are we ever going to support the py2 `exec` statement?